utils/common/src/main/kotlin/EnvironmentVariableFilter.kt

```diff
@@ -107,6 +113,10 @@ object EnvironmentVariableFilter {
     fun filter(environment: MutableMap<String, String>): MutableMap<String, String> {
         val keysToRemove = environment.keys.filterNot(EnvironmentVariableFilter::isAllowed)
 
+        if (keysToRemove.isNotEmpty()) {
+            logger.debug { "Filtering out these variables from environment: $keysToRemove." }
```

Comment:
Nits:
- "from *the* environment"
- Here, I'd drop the dot at the end, as you're printing list contents after a ":" which stands on its own.